src/createElementFunction.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Relay from 'react-relay';
+
+import * as Utils from './containerUtils';
+import QueryAggregator from './QueryAggregator';
+import getRouteParams from './getRouteParams';
+import RouteGenerator from './RouteGenerator';
+
+export function createElementFunction() {
+  const queryAggregator = new QueryAggregator();
+  const routeGenerator = new RouteGenerator();
+
+  return function createElement(Component, props) {
+    const {branch, route, params, location, routeParams} = props;
+
+    // Wrap any Relay.Containers
+    let element = !Relay.isContainer(Component) ? <Component {...props} /> :
+      Utils.createContainerElement(
+        Component,
+        props,
+        getRouteParams(routeParams, location.query, route.queryParams || []),
+        route.rootContainerProps || {},
+        routeGenerator.getRouteFor(branch),
+        queryAggregator
+      );
+
+    // Wrap the root component in a RootContainer
+    element = route !== branch[0] ? element :
+      Utils.createBatchedRelayContainer(
+        element,
+        getRouteParams(params, location.query, route.queryParams || []),
```

Comment:
This is actually not quite right - this is looking only at the allowed `queryParam` on the root `Route`, but it should actually be the union of all the query params down the entire route tree - otherwise we drop some query params.